### ISSUE TITLE
SPIKE: Romanization Search - Unicode and Non-Roman

### ISSUE THREAD

**Issue by lovinscari** [emory-libraries/blacklight-catalog#99]
As an end user, I want the ability to search using Unicode and non-roman characters contained in metadata. So if I have a method of entering non-roman character words or text strings, I can get back results that match records in the index without having to know how a "romanized" version might be represented. Because it is unclear what is involved in searching the data, a Spike ticket is needed to investigate how Solr and Blacklight handles searches of Unicode and Non-Roman Characters. If it is determined modifications are needed to Solr and/or Blacklight, those details will be documented in new tickets.

- Examples of Unicode Characters 
- Examples of Non-Roman Characters
- Examples of searching of Unicode Characters (Bib ID)
- Examples of searching of Non-Roman Characters (Bib ID)

From Laura (see comments below):

Search term in Arabic:
جودة الفواكة

One result: MMSID 9937110709802486
Title: جودة الفواكة والخضار وطرق تجفيفها = Fruit and Vegetable Quality and their Drying Methods

Example where Arabic Unicode characters and English roman alphabet are combined in one title field with “parallel titles” and there are 880 fields with the Arabic script and Romanized Arabic script versions of several fields

Search term in Chinese:
商

133 results including MMSID 9936494528102486
Title: 商訊

Very few Latin script words displaying in the record

Search term in Hebrew:
אוסלו

5 results including: MMSID 9936997911202486
Title:  Medinat Yiśraʼel le-an? :$betgarim li-zehutah ha-Yehudit ṿeha-demoḳraṭit shel Medinat Yiśraʼel u-mitṿeh le-hitmodedut ʻimam

Includes a Table of Contents only in Hebrew.

LINK TO METADATA MODEL - https://docs.google.com/spreadsheets/d/1jSqgNd6HnVJdJMrrShAofkmK8uGgHGdlKODmZOL5NfY/edit?usp=sharing

**Comment by lovinscari** [comment]
@laura-ake - Can you please provide examples indicated in the issue?

**Comment by laura-ake** [comment]
I have 3 examples - am testing whether I can paste them in here and the Unicode chars. will show up:

Search term in Arabic:
جودة الفواكة

One result: MMSID 9937110709802486 
Title:  جودة الفواكة والخضار وطرق تجفيفها = Fruit and Vegetable Quality and their Drying Methods
 
Example where Arabic Unicode characters and English roman alphabet are combined in one title field with “parallel titles” and there are 880 fields with the Arabic script and Romanized Arabic script versions of several fields

Search term in Chinese:
商

133 results including MMSID 9936494528102486
Title:  商訊

Very few Latin script words displaying in the record


Search term in Hebrew:  
אוסלו

5 results including: MMSID 9936997911202486
Title:  

Includes a Table of Contents only in Hebrew.

Yay, looks like it's good.  Had to use Notepad+ to gather, Word was doing weird things when I was pasting.

@lovinscari @eporter23  is this enough for now or do you want more?  I will try to find records with special Unicode symbol characters as well and will write Tarina, Sofia and Guo-Hua to solicit more.

**Comment by eporter23** [comment]
@lovinscari per @laura-ake 's comment above, perhaps another ticket for additional test cases may be helpful in the future.

**Comment by bwatson78** [comment]
Initial Findings:

**Arabic: جودة الفواكة (MMS id: 9937110709802486)**
- I did not find this item in the current Blacklight/Solr catalog by either id or unicode text.
- Spoke to @lisahamlett, and she confirms that it should be a part of the catalog, but it's being denied indexing.

**Chinese: 商訊 (MMS id: 9936494528102486)**
- I searched the above title whole and by first character and the application couldn't find it.
- I searched by id and was able to find the record, but I then copied/pasted the title into the search field and couldn't find it in either Title or All Fields.

**Hebrew: אוסלו (MMS id: 9936997911202486)**
- This produced zero results by both title and id.
- @lisahamlett confirmed that this should have been indexed into our system as well. She posted the MARC records on Slack for me to analyze. That's my next step.

**Comment by bwatson78** [comment]
@lovinscari 
@lisahamlett made a OAI set that included just those two. I harvested them and have figured out why they're not showing in any of my searches. So far, Keyword is the only search configuration we have in the `main` branch. Keyword isn't looking in any of the fields that contain the Unicode/Non-Roman titles. Since Zo has been sick, we don't have the code for the Title search settings.

Can I recommend that we put this ticket in Blocked and I churn out the changes needed for Title and, if you think it prudent, expand out the fields that Keyword looks at?

**Comment by laura-ake** [comment]
@lovinscari @bwatson78 ,
Here is the MMSID for a record with Chinese characters 中國 (romanized Zhongguo) in 245 title field which is indexed:  9936494871702486.  It's in blackcat-arch and is findable by:

- 9936494871702486 in Keyword search and
"中國金融 =" in Title search  (note quote marks around the title string - necessary because of the problem Devanshu reported yesterday that keywords are not being searched as words, but quoted strings can be searched)

It is not found by

"中國金融 =" in Keyword search, because the Title indexed fields have not been added to Keyword
"中國" in Title search, not sure why.   That search finds record 990010391520302486 (Title showing in display is just "Zhongguo")

The 245 field for the starting example:
=245  00$a中國金融 =$bChina finance    .  This is a "parallel title" with the additional title found on the title page in subfield b.  This should be displayed and indexed, isn't displaying or apparently being indexed (Title search on "China finance" failed) right now.

The Primo record:  https://discovere.emory.edu:443/discovere:default_scope:01EMORY_ALMA51272033160002486 

I think being able to demonstrate this ticket is blocked by Devanshu's reported problem with indexing (#128?)  

Brad, need more examples?

**Comment by lovinscari** [comment]
@bwatson78 - I received more information yesterday around the Marc fields associated with the title_display and title_vern_display that indicate the need to have the 880 field mapped with our title search, which might be why some of the items identified in this user story are not appearing as we anticipated. I will update the ticket with the link to the metadata model and change the search specifically to title as I believe we now have enough information to complete a title search successfully with this use case.

**Comment by lovinscari** [comment]
@bwatson78 See above for a link to the metadata model. I will also provide a link to the issue I am creating with the read me file, etc.

**Comment by bwatson78** [comment]
@lovinscari I have just searched and found the Arabic and Hebrew items listed above after #128 was merged into production. Screenshots below:

![Screen Shot 2021-01-19 at 4.12.32 PM.png](https://images.zenhubusercontent.com/5e6f6c00c3241244877b4aea/608925f9-2134-4a4c-b562-7eec38054f13)
![Screen Shot 2021-01-19 at 4.18.16 PM.png](https://images.zenhubusercontent.com/5e6f6c00c3241244877b4aea/03f62e24-34f6-4186-967e-74c90eb5356e)
The original Chinese item was not in this current OAI set, but was able to find a Harry Potter book in Chinese using a partial title.

![Screen Shot 2021-01-19 at 4.06.54 PM.png](https://images.zenhubusercontent.com/5e6f6c00c3241244877b4aea/dad074df-45e9-4860-b8fb-3a67341a2027)
